app/viewmodels/SecurityStatementsViewModel.scala

```diff
@@ -16,9 +16,259 @@
+                                       header: HtmlFormat.Appendable,
+                                       requestedStatementNotification: HtmlFormat.Appendable,
+                                       currentStatements: HtmlFormat.Appendable,
+                                       missingGuidance: HtmlFormat.Appendable,
+                                       statementServiceParagraph: HtmlFormat.Appendable,
+                                       requestStatementsLink: HtmlFormat.Appendable,
+                                       details: SecurityStatementsViewModelDetails)
+
+case class SecurityStatementsViewModelDetails(statementsForAllEoris: Seq[SecurityStatementsForEori],
+                                              hasRequestedStatements: Boolean,
+                                              hasCurrentStatements: Boolean)
+
+object SecurityStatementsViewModel {
+  def apply(statementsForAllEoris: Seq[SecurityStatementsForEori])(implicit appConfig: AppConfig,
+                                                                   messages: Messages): SecurityStatementsViewModel = {
+    SecurityStatementsViewModel(
+      pageTitle = Some(messages("cf.security-statements.title")),
+      backLink = Some(appConfig.customsFinancialsFrontendHomepage),
+      header = generateHeader,
+      requestedStatementNotification = generateRequestedStatementNotification(statementsForAllEoris),
+      currentStatements = generateCurrentStatements(statementsForAllEoris),
+      missingGuidance = generateMissingGuidance,
+      statementServiceParagraph = generateStatementServiceParagraph,
+      requestStatementsLink = generateRequestStatementLink,
+      details = SecurityStatementsViewModelDetails(
+        statementsForAllEoris = statementsForAllEoris,
+        hasRequestedStatements = hasRequestedStatements(statementsForAllEoris),
+        hasCurrentStatements = hasCurrentStatements(statementsForAllEoris)))
+  }
+
+  private def generateHeader(implicit messages: Messages): HtmlFormat.Appendable = {
+    h1Component(msg = "cf.security-statements.title", classes = "govuk-heading-xl")
+  }
+
+  private def generateRequestedStatementNotification(statementsForAllEoris: Seq[SecurityStatementsForEori])
+                                                    (implicit appConfig: AppConfig, messages: Messages
+                                                    ): HtmlFormat.Appendable = {
+    if (hasRequestedStatements(statementsForAllEoris)) {
+      new requestedStatements(linkComponent).apply(
+        url = appConfig.requestedStatements(SecurityStatement))
+    } else {
+      HtmlFormat.empty
+    }
+  }
+
+  private def generateCurrentStatements(statementsForAllEoris: Seq[SecurityStatementsForEori])
+                                       (implicit messages: Messages): HtmlFormat.Appendable = {
+    if (hasCurrentStatements(statementsForAllEoris)) {
+      HtmlFormat.fill(statementsForAllEoris.zipWithIndex.flatMap {
```

Comment:
Could you please move this functionality into a separate method to make is more readable like below? I chose a method name, check if that is meaningful for you but we need to keep a name on the same line to understand the flow easily

  private def generateCurrentStatements(statementsForAllEoris: Seq[SecurityStatementsForEori])
                                       (implicit messages: Messages): HtmlFormat.Appendable = {
    if (hasCurrentStatements(statementsForAllEoris)) {
      HtmlFormat.fill(populateCurrentPdfAndCsvStatements(statementsForAllEoris))
    } else {
      generateNoStatementsParagraph
    }
  }

  private def populateCurrentPdfAndCsvStatements(statementsForAllEoris: Seq[SecurityStatementsForEori])
                                                (implicit messages: Messages): Seq[HtmlFormat.Appendable] = {
    statementsForAllEoris.zipWithIndex.flatMap {
      case (statementsForEori, historyIndex) if statementsForEori.currentStatements.nonEmpty =>
        val eoriHeader = generateEoriHeader(historyIndex, statementsForEori)
        val eomHeader = generateEomHeader(statementsForEori)

        val pdfStatements = generateStatements(statementsForEori, historyIndex, isCsv = false)
        val csvStatements = generateStatements(statementsForEori, historyIndex, isCsv = true)

        Seq(eoriHeader, Some(pdfStatements), eomHeader, Some(csvStatements)).flatten

      case _ => Seq.empty
    }
  }

Reply:
This has been addressed in the commit [73dfe78](https://github.com/hmrc/customs-financials-documents-frontend/pull/101/commits/73dfe781532fd8bddb0967c9eeb15974440beaf7)